### PR TITLE
fix: Days left are wrongly displayed for challenges - MEED-1776

### PR DESCRIPTION
**Pull request by SaraBoutej**
Fix Feedback related to the numbers of left days in the challenge card.

### CODE REVIEW

File: portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue [hunk 1]
```diff
@@ -171,7 +171,7 @@ export default {
         const days = Math.round((this.startDate.getTime() - new Date().getTime()) / (1000 * 60 * 60 * 24)) + 1;
         return this.$t('challenges.label.openIn', {0: days});
       } else {
-        const days = Math.round((this.endDate.getTime() - this.startDate.getTime()) / (1000 * 60 * 60 * 24)) + 1;
+        const days = Math.round((this.endDate.getTime() - new Date().getTime()) / (1000 * 60 * 60 * 24)) + 1;
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue:174]: please replace dates comparaison in previous `if else`, the date comparaison should be done using timestamp comparaison, try `(new Date() > new Date()) || (new Date() < new Date()) || (new Date() == new Date())` it never gives `true`

Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue:174]: cc @boubaker 
When I try to put this code, an eslint error appears "no-self-compare"

image

Comment (boubaker) [reply portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue:174]: I didn't meant to add the code inside the file, it's just to indicate that the dates comparaison doesn't work and you should use Date.getTime for timestamp comparaison. The given code is to test on browser console if you want to see that it never gives true as said.

Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue:174]: cc @boubaker done!
I misunderstood what you said, sorry!

**Review by boubaker — DISMISSED** [review]

File: portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue [hunk 2]
```diff
@@ -167,11 +167,11 @@ export default {
     remainingPeriodLabel() {
       if (this.endDate < new Date()) {
```
Comment (boubaker) [inline portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue]: same here

File: portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue (hunk 1 above)
Comment (boubaker) [reply portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue:174]: no worry, thanks for considering

File: portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue (hunk 2 above)
Comment (SaraBoutej) [reply portlets/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeCard.vue]: Done!

**Review by boubaker — APPROVED** [review]

**Pull request merged**